File: aoj/volume0/n0021/Main.cpp
#include <iostream>

using namespace std;

int main() {

    int n;
    cin >> n;

    for (int i = 0; i < n; i++) {

        float x1, y1, x2, y2, x3, y3, x4, y4;
        cin >> x1 >> y1 >> x2 >> y2 >> x3 >> y3 >> x4 >> y4;

        float slope1 = (y1 - y2) / (x1 - x2);
        float slope2 = (y3 - y4) / (x3 - x4);

        if (slope1 == slope2) {
            cout << "YES" << endl;
        } else {
            cout << "NO" << endl;
        }

    }

    return 0;

}

File: aoj/volume0/n0021/Main.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int N = sc.nextInt();

        for (int i = 0; i < N; i++) {

            double aX = sc.nextDouble();
            double aY = sc.nextDouble();
            double bX = sc.nextDouble();
            double bY = sc.nextDouble();

            double cX = sc.nextDouble();
            double cY = sc.nextDouble();
            double dX = sc.nextDouble();
            double dY = sc.nextDouble();

            double slope1 = (bY - aY) / (bX - aX);
            double slope2 = (dY - cY) / (dX - cX);

            System.out.println(slope1 == slope2 ? "YES" : "NO");

        }

    }

}
